tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,307 @@
+                       [CandID] => " . $candID . " where there are dates.");
+        }
+        // Check if consent status is yes but consent date is empty
+        if($status === "yes" && empty($date)) {
+            array_push($errors, "The date of consent is missing for " . $consentName . ":
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       All 'yes' statuses must have a consent date.");
+        }
+        // Check if consent status is no and consent date given, but withdrawal date is empty
+        if($status === "no" && !empty($date)) {
+            if(empty($withdrawal)) {
+                array_push($errors, "The date of withdrawal is missing for " . $consentName . ":
+                           [ID]     => " . $consentID . "
+                           [CandID] => " . $candID . "
+                           All 'no' statuses with given consent date must have a withdrawal date.");
+            }
+        }
+        // Check for zero dates
+        if($date === "0000-00-00" || $withdrawal === "0000-00-00") {
+            array_push($errors, "Zero dates found in participant_status for " . $consentName . ":
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       Please remove date or run /tools/DB_date_zeros_removal.php.");
+        }
+    }
+}
+////////////////////////////////////////////////
+// CHECK NEW TABLES AND CONFIG SETTINGS EXIST //
+////////////////////////////////////////////////
+
+// Check useConsent was inserted into ConfigSettings
+try {
+    $configID = $db->pselectOne("SELECT ID FROM ConfigSettings WHERE Name='useConsent'", array());
+} catch (DatabaseException $e) {
+    array_push($errors, "useConsent missing from ConfigSettings table.
+               Run SQL/New_patches/2018-03-01_normalise_consent.sql");
+}
+try {
+    $configValue = $db->pselectOne("SELECT Value FROM Config WHERE ConfigID=:configid",
+                       array('configid'=>$configID));
+} catch (DatabaseException $e) {
+    array_push($errors, "useConsent's Config value missing from Config table.
+               Run SQL/New_patches/2018-03-01_normalise_consent.sql");
+}
+if (!$db->tableExists('consent') || !$db->tableExists('candidate_consent_rel') || !$db->tableExists('candidate_consent_history')) {
+    array_push($errors, "New, normalized consent tables do not exist.
+               Run SQL/New_patches/2018-03-01_normalise_consent.sql");
+}
+//////////////////
+// THROW ERRORS //
+//////////////////
+if (!empty($errors)) {
+    print_r($errors);
+    echo "\nResolve errors and run script again.\n";
+    die();
+} else {
+    echo "\nValidation successful.\n";
+}
+/////////////////////
+// CONTINUE SCRIPT //
+/////////////////////
+
+// Update ConfigSetting table with value of 'useConsent' if true. Default is set to false.
+if ($useConsent === "true") {
```

Comment:
I think this can also be "1" in some `config.xml` files

Reply:
hmm okay